File: CommandsList.java
import java.util.Dictionary;

public class CommandsList
{
    // The reference to the main program, so we can access other classes, like Output
    private Program referenceToProgram;
    // The list of all commands
    public Dictionary<String, Command> listOfCommands = new HashTable();
    
    public CommandsList(Program program)
    {
        referenceToProgram = program;
    }
}

File: CommandsList.cpp
#include "CommandsList.h"
using namespace std;

CommandsList::CommandsList(Program* program)
{
    referenceToProgram = program;

    
    // When adding new commands, they need to be inserted into the listOfCommands here

    //listOfCommands.insert("go", new CommandGo(referenceToProgram));
}